src/repositories/request-repository.ts

```diff
@@ -21,15 +21,35 @@ export const FAILURE_RETRY_WINDOW = 1000 * 60 * 60 * 48 // 48H
 const REPEATED_READ_SERIALIZATION_ERROR = '40001'
 export const TABLE_NAME = 'request'
 
-const countRetryMetrics = (requests: Request[], anchoringDeadline: Date): void => {
-  const expired = requests.filter((request) => request.createdAt < anchoringDeadline)
-  if (expired.length > 0) Metrics.count(METRIC_NAMES.RETRY_EXPIRING, expired.length)
-
-  const processing = requests.filter((request) => request.status === RequestStatus.PROCESSING)
-  if (processing.length > 0) Metrics.count(METRIC_NAMES.RETRY_PROCESSING, processing.length)
+/**
+ * Records statistics about the set of requests
+ * Groups by EXPIRED (if the time is past the deadline), PROCESSING, and FAILED
+ * 
+ * Will record the total count, the mean time since createdAt, and the max time since createdAt
+ * for each group
+ *
+ * @param requests
+ * @param anchoringDeadline
+ * @returns
+ */
+const recordAnchorRequestMetrics = (requests: Request[], anchoringDeadline: Date): void => {
+  const expired = new TimeableMetric(SinceField.CREATED_AT)
+  const processing = new TimeableMetric(SinceField.CREATED_AT)
+  const failed = new TimeableMetric(SinceField.CREATED_AT)
+
+  for (const req of requests) {
+      if (req.createdAt < anchoringDeadline) {
+          expired.record(req)
+      } else if (req.status === RequestStatus.PROCESSING) {
```

Comment:
ok!  I wasn't sure if we should be counting in both categories, but now we are, so we will have the timing including expirations. that's probably good since we need to know if they are taking a long time, which category they are in